service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -58,8 +58,18 @@ readonly TERRA_SERVER=$(get_metadata_value "instance/attributes/terra-cli-server
 if [[ -n "${TERRA_SERVER}" ]]; then
   sudo -u "${JUPYTER_USER}" sh -c "terra server set --name=${TERRA_SERVER}"
 fi
+
 # Set the CLI terra workspace id using the VM metadata, if set.
 readonly TERRA_WORKSPACE=$(get_metadata_value "instance/attributes/terra-workspace-id")
 if [[ -n "${TERRA_WORKSPACE}" ]]; then
   sudo -u "${JUPYTER_USER}" sh -c "terra workspace set --id=${TERRA_WORKSPACE} --defer-login"
 fi
+
+# Log in with app-default-credentials
+sudo -u "${JUPYTER_USER}" sh -c "terra auth login --mode=APP_DEFAULT_CREDENTIALS"
+
+sudo -u "${JUPYTER_USER}" sh -c "terra resource list --type=GIT_REPO" | awk '{print $1}' | while read line; do
+  sudo -u "${JUPYTER_USER}" sh -c "terra resource describe --name=$line" | awk '/Git repo Url/ {print $4}'
```

Comment:
I'm having trouble following this, as I'm not a bash or `awk` expert. Also, why are you using the text output instead of JSON? Seems like JSON would be easier to parse (if you've got `jq` on the instance, which if we don't, we should IMO).

Reply:
oh i didn't think about jSON. the instance have jq. I will try to simplify this.

Reply:
use `jq` which simplify quiet a bit.